render.go

```diff
@@ -152,7 +151,7 @@ func (ren *RenderComponent) generateBufferContent() []float32 {
 	red := colorR
 	green := colorG << 8
 	blue := colorB << 16
-	alpha := uint32(transparency*255.0) << 24
+	alpha := uint32(ren.Transparency*255.0) << 24
```

Comment:
Personally, I am fine with the current implementation - Regenerating the buffer seems alright to me. Realistically, I don't think it will be happening that often. If it becomes an issue in the future, changing it up should not be too much of an issue.